src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcSocketsClient.cs

```diff
@@ -111,42 +107,64 @@ private void IncrementBytesSentMetric(int size)
         }
 
         public virtual async Task<int> SendJsonRpcResult(JsonRpcResult result)
+        {
+            if (result.IsCollection)
+            {
+                int singleResponseSize = 0;
+                bool isFirst = true;
+                await _handler.SendRawAsync(_jsonOpeningBracket);
+                await foreach (JsonRpcResult.Entry innerResult in result.BatchedResponses!)
+                {
+                    if (!isFirst) await _handler.SendRawAsync(_jsonComma);
```

Comment:
If we dig deeper, we find:
- In `IpcSocketsHandler`, `_socket.SendAsync(data, SocketFlags.None)`, not sure if we should set `SocketFlags.Truncated` or `SocketFlags.Partial` until we send `_jsonClosingBracket`.
- In `WebSocketHandler` we have `_webSocket.SendAsync(data, WebSocketMessageType.Text, true, CancellationToken.None)`, maybe we should set `endOfMessage` to `false` until we send `_jsonClosingBracket`.